src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -1726,8 +1714,83 @@ ValidPermissionedDomain::finalize(
         return true;
     };
 
-    return (sleStatus_[0] ? check(*sleStatus_[0], j) : true) &&
-        (sleStatus_[1] ? check(*sleStatus_[1], j) : true);
+    if (view.rules().enabled(fixPermissionedDomainInvariant))
+    {
+        // Nothing was changed by a failed tx. All good.
+        if (result != tesSUCCESS && sleStatus_.size() == 0)
+            return true;
+
+        if (sleStatus_.size() > 1)
+        {
+            JLOG(j.fatal())
+                << "Invariant failed: transaction affected more than 1 entry.";
```

Comment:
Just to make the error message clearer:

```suggestion
            JLOG(j.fatal())
                << "Invariant failed: transaction affected more than 1 permissioned domain entry.";
```